cpp/src/arrow/util/async_generator.h

```diff
@@ -20,6 +20,7 @@
 #include <cassert>
 #include <deque>
 #include <queue>
+#include <thread>
```

Comment:
This is unfortunate but we could easily add a method for getting the current thread id into thread_pool.h once ARROW-12878 is merged.